sprint7/encyklopedia1.js

```diff
@@ -0,0 +1,11 @@
+/*Constructor function
+Słowo kluczowe “this” w kontekście funkcji*/
+
+/*
+Constructor function to zwykła funkcja, pisana z wielkiej litery. Aby utworzyć nowe obiekty na bazie konstruktora skorzystamy ze słowa kluczowego 'new'. Gdy funkcja jest wykonywana za pomocą 'new', wykonuje następujące kroki:
+1. nowy pusty obiekt zostanie utworzony i przypisany do 'this'.
+2. wykonuje się ciało funkcji. 'This' dodaje do niego nowe właściwości.
+3. wartość 'this' zwraca.
+
+Słowo kluczowe 'this' moze uzyskać dostęp do obiektu. Wartością 'this' jest obiekt "przed kropką", uzywany do wywołania funkcji.
+*/
```

Comment:
Pani Ewo w trochę innej formie spróbujmy:

CONSTRUCTOR FUNCTION:
1) Co to jest constructor function? (Pani zdanie jest ok)
2) Po co tworzymy constructor function?
3) Jaką konwencję warto zastosowac przy tworzeniu constructor function?
4) Jaka jest różnica między constructor function a zwykłą funkcją ze względu na:
   a) słowo kluczowe `this`? Na co wskazuje this w constructor function a na co w zwykłej?
   b) return - co zwraca constructor function a co zwykła?
Proszę zrobić od nowa, skasować swój opis i dodać nowy opierając się powyższych punktach :)

SŁOWO KLUCZOWE “THIS” W KONTEKŚCIE FUNKCJI:
To tak naprawdę jest punkt 4 powyżej.

Więc jeśli pani dobrze opisze CONSTRUCTOR FUNCTION to nie musi pani robić SŁOWO KLUCZOWE “THIS” W KONTEKŚCIE FUNKCJI :)